modules/mocked_authentication/app/controllers/mocked_authentication/credential_providers_controller.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module MockedAuthentication
+  class CredentialProvidersController < ApplicationController
+    skip_before_action :authenticate
+
+    def authorize
+      parsed_info = parse_credential_info(params[:credential_info])
+      credential_info_code = SecureRandom.hex
+      MockCredentialInfo.new(credential_info_code: credential_info_code, credential_info: parsed_info).save!
+
+      render json: { credential_info_code: credential_info_code }
+    rescue => e
+      render json: { errors: e }, status: :bad_request
```

Comment:
At first I was gonna ask if we could make the error response closer to that of the SigninController. But actually I think this is better. For mocked auth it's better to have a hard fail and let it be super easy for devs to know what's going on